templates/vue/src/components/TapestryMap.vue

```diff
@@ -0,0 +1,324 @@
+        >
+          Add to map
+        </b-button>
+      </b-card>
+      <b-button class="add-new-node-btn" block @click="addNewNode">
+        <i class="fas fa-plus-circle fa-3x"></i>
+        <br />
+        Add new node
+      </b-button>
+    </div>
+    <div class="map-content">
+      <l-map
+        ref="map"
+        :options="mapOptions"
+        :bounds="getBounds"
+        :max-bounds="worldBounds"
+        @update:center="updateCenter"
+        @update:zoom="updateZoom"
+      >
+        <l-tile-layer :url="url" :attribution="attribution" />
+        <l-marker
+          v-for="marker in markerlocations"
+          :key="marker.id"
+          :ref="'marker-' + marker.id"
+          :lat-lng="marker.pos"
+          :icon="
+            marker.status == 'draft'
+              ? draftIcon
+              : marker.accessible
+              ? icon
+              : inaccessibleIcon
+          "
+          @click="selectNode(marker.id)"
+        >
+          <l-popup>
+            <b-button
+              v-if="marker.accessible || canEdit"
+              variant="link"
+              @click="openNode(marker.id)"
+            >
+              <h6>{{ marker.title }}</h6>
+            </b-button>
+            <div v-else>
+              <h6>{{ marker.title }}</h6>
+            </div>
+          </l-popup>
+        </l-marker>
+      </l-map>
+    </div>
+  </div>
+</template>
+
+<script>
+import "leaflet/dist/leaflet.css"
+import mapMarker from "@/assets/map-marker.png"
+import invalidMarker from "@/assets/map-marker-inaccessible.png"
+import draftMarker from "@/assets/map-marker-draft.png"
+import { latLng, latLngBounds, icon } from "leaflet"
+import { LMap, LTileLayer, LMarker, LPopup } from "vue2-leaflet"
+import { mapState } from "vuex"
+import { names } from "@/config/routes"
+import Helpers from "@/utils/Helpers"
+import { isLoggedIn } from "@/utils/wp"
+
+export default {
+  name: "tapestry-map",
+  components: {
+    LMap,
+    LTileLayer,
+    LMarker,
+    LPopup,
+  },
+  props: {
+    isSidebarOpen: {
+      type: Boolean,
+      required: false,
+    },
+  },
+  data() {
+    return {
+      url: "https://{s}.tile.openstreetmap.org/{z}/{x}/{y}.png",
+      attribution:
+        '&copy; <a href="http://osm.org/copyright">OpenStreetMap</a> contributors',
+      mapOptions: {
+        zoomSnap: 0.1,
+        scrollWheelZoom: false,
+      },
+      icon: icon({
+        iconUrl: mapMarker,
+        iconSize: [32, 33],
+        iconAnchor: [16, 32],
+        popupAnchor: [0, -20],
+      }),
+      inaccessibleIcon: icon({
+        iconUrl: invalidMarker,
+        iconSize: [32, 33],
+        iconAnchor: [16, 32],
+        popupAnchor: [0, -20],
+      }),
+      draftIcon: icon({
+        iconUrl: draftMarker,
+        iconSize: [32, 33],
+        iconAnchor: [16, 32],
+        popupAnchor: [0, -20],
+      }),
+      worldBounds: latLngBounds([
+        [-90, -180],
+        [90, 180],
+      ]),
```

Comment:
Do these attributes change? Typically `data` attributes are reserved for those that change throughout the component lifecycle. Otherwise you would put them under `computed`. Part of the reason is performance, but it also shows people who are reading the code that it won't change.